PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/actions/CreateFunctionAction.java

```diff
@@ -0,0 +1,214 @@
+import com.intellij.ide.actions.CreateElementActionBase;
+import com.intellij.ide.actions.CreateFileAction;
+import com.intellij.ide.highlighter.JavaFileType;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.actionSystem.DataContext;
+import com.intellij.openapi.actionSystem.LangDataKeys;
+import com.intellij.openapi.application.Application;
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.command.CommandProcessor;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.roots.ProjectFileIndex;
+import com.intellij.openapi.roots.ProjectRootManager;
+import com.intellij.openapi.util.Computable;
+import com.intellij.openapi.util.IconLoader;
+import com.intellij.openapi.util.text.StringUtil;
+import com.intellij.psi.JavaDirectoryService;
+import com.intellij.psi.PsiDirectory;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.PsiFileFactory;
+import com.intellij.psi.PsiNameHelper;
+import com.intellij.psi.PsiPackage;
+import com.microsoft.azure.common.exceptions.AzureExecutionException;
+import com.microsoft.azure.common.function.template.FunctionTemplate;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.eventhub.EventHubNamespace;
+import com.microsoft.azure.management.eventhub.EventHubNamespaceAuthorizationRule;
+import com.microsoft.azuretools.authmanage.AuthMethodManager;
+import com.microsoft.intellij.forms.function.CreateFunctionForm;
+import com.microsoft.intellij.runner.functions.AzureFunctionSupportConfigurationType;
+import com.microsoft.intellij.util.AzureFunctionsUtils;
+import com.microsoft.intellij.util.PluginUtil;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.jps.model.java.JavaModuleSourceRootTypes;
+
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Writer;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class CreateFunctionAction extends CreateElementActionBase {
+    public CreateFunctionAction() {
+        super("Azure Function Class",
+                "newPage.dialog.prompt", IconLoader.getIcon(AzureFunctionSupportConfigurationType.ICON_PATH,
+                        AzureFunctionSupportConfigurationType.class));
+    }
+
+    @Override
+    protected PsiElement[] invokeDialog(Project project, PsiDirectory psiDirectory) {
+        PsiPackage pkg = JavaDirectoryService.getInstance().getPackage(psiDirectory);
+        // get existing package from current directoy
+        String hintPackageName = pkg == null ? "" : pkg.getQualifiedName();
+
+        CreateFunctionForm form = new CreateFunctionForm(project, hintPackageName);
```

Comment:
Shall we move the create logic outer the Action class?

Reply:
Sure, we will move the create project logic to external for create function project wizard and create function class wizard. It will be done in later PR